lsp.go

```diff
@@ -230,59 +225,69 @@ func (s *Server) handleDocumentSaved(
 	}
 
 	fname := params.TextDocument.URI.Filename()
-	data, err := os.ReadFile(fname)
+	content, err := os.ReadFile(fname)
 	if err != nil {
 		log.Error().Err(err).Msg("reading saved file.")
 		return nil
 	}
 
-	err = checkFile(fname, string(data))
-	return reply(ctx, nil,
-		s.sendErrorDiagnostics(ctx, params.TextDocument.URI, err),
-	)
+	return s.checkAndReply(ctx, reply, fname, string(content))
 }
 
-func (s *Server) sendErrorDiagnostics(ctx context.Context, currentFile lsp.URI, err error) error {
-	if err == nil {
-		log.Debug().Str("file", currentFile.Filename()).
-			Msg("cleaning editor errors")
-
-		// this is required to clear the `problems panel` for the active file
-		// if it had errors before.
-		s.sendDiagnostics(ctx, currentFile, []lsp.Diagnostic{})
-		return nil
+// sendErrorDiagnostics sends diagnostics for each provided file, the ones with
+// no reported error gets an empty list of diagnostics, so the editor can clean
+// up its problems panel for it.
+func (s *Server) sendErrorDiagnostics(ctx context.Context, files []string, err error) error {
+	errs := errors.L()
+	switch e := err.(type) {
+	case *errors.Error:
+		// an error can wrap a list of errors.
+		errs = e.AsList()
+	case *errors.List:
+		errs = e
+	default:
+		if err != nil {
+			log.Debug().Err(err).Msg("unknown error ignored because it doesn't provide file range")
+		}
 	}
 
-	e, ok := err.(*errors.Error)
-	if !ok {
-		log.Debug().Err(err).Msg("unknown error ignored because it doesn't provide file range")
-		return nil
+	diagsMap := map[string][]lsp.Diagnostic{}
+	for _, filename := range files {
+		if _, ok := diagsMap[filename]; !ok {
+			diagsMap[filename] = []lsp.Diagnostic{}
+		}
 	}
```

Comment:
```suggestion
	for _, filename := range files {
			diagsMap[filename] = []lsp.Diagnostic{}
	}
```

Rationale: we just created the diagsMap, so it will always be !ok no ? (when I mentioned using !ok I was thinking later on the method, near the end, so we would handle any gaps)

Reply:
yep, bad copy pasting. This was handled after going over the errors, and then it needed the check to see if it had errors or not.